dynamiq/serializers/dumpers/yaml.py

```diff
@@ -0,0 +1,202 @@
+import enum
+from os import PathLike
+from typing import IO, Any
+
+from dynamiq.connections import BaseConnection
+from dynamiq.flows import Flow
+from dynamiq.nodes import Node
+from dynamiq.serializers.types import WorkflowYamlData
+from dynamiq.utils.logger import logger
+
+
+class WorkflowYAMLDumperException(Exception):
+    pass
+
+
+class WorkflowYAMLDumper:
+    """Dumper class for parsing workflow components and save them to YAML."""
+
+    @classmethod
+    def get_updated_node_data(
+        cls,
+        node_data: dict,
+        connections_data: dict[str, dict],
+        skip_nullable: bool = False,
+    ):
+        """
+        Get node init data with and connections components recursively (llms, agents, etc)
+
+        Args:
+            node_data: Dictionary containing node data.
+            connections_data: Dictionary containing connections shared data.
+            skip_nullable: Skip nullable fields.
+
+        Returns:
+            A dictionary of newly created nodes with dependencies.
+        """
+        updated_node_init_data = {}
+        for param_name, param_data in node_data.items():
+            if param_name == "depends":
+                updated_node_init_data[param_name] = [
+                    {"node": dep["node"]["id"], "option": dep["option"]} for dep in param_data
+                ]
+
+            elif param_name == "connection":
```

Comment:
What about `prompt`? Or are we always going to dump it inline?

Reply:
Currently just supports inline prompts.